cmd/postcli/main.go

```diff
@@ -102,75 +100,144 @@ func askForConfirmation() {
 		return
 	}
 
-	fmt.Println("Are you sure you want to continue (y/N)?")
+	log.Println("Are you sure you want to continue (y/N)?")
 
 	var answer string
 	_, err := fmt.Scanln(&answer)
 	if err != nil || !(answer == "y" || answer == "Y") {
-		fmt.Println("Aborting")
-		os.Exit(1)
+		log.Fatal("Aborting")
 	}
 }
 
-func processFlags() error {
+func processFlags() {
+	meta, err := initialization.LoadMetadata(opts.DataDir)
+	switch {
+	case errors.Is(err, initialization.ErrStateMetadataFileMissing):
+	case err != nil:
+		log.Fatalln("failed to load metadata:", err)
+	default:
+		if idHex == "" {
+			idHex = hex.EncodeToString(meta.NodeId)
+		} else if idHex != hex.EncodeToString(meta.NodeId) {
+			log.Println("WARNING: it appears that", opts.DataDir, "was previously initialized with a different `id` value.")
+			log.Println("\tCurrent value:", hex.EncodeToString(meta.NodeId))
+			log.Println("\tValue passed to postcli:", idHex)
+			if !yes {
+				log.Println("CONTINUING MIGHT ERASE EXISTING DATA. MAKE ABSOLUTELY SURE YOU SPECIFY THE CORRECT VALUE.")
+			}
+			askForConfirmation()
+		}
+	}
+
+	key, err := loadKey()
+	switch {
+	case errors.Is(err, os.ErrNotExist):
+	case err != nil:
+		log.Fatalln("failed to load key:", err)
+	case meta != nil && !bytes.Equal(meta.NodeId, key):
+		log.Fatalln("WARNING: inconsistent state:", edKeyFileName, "file does not match metadata in", opts.DataDir)
+	default:
+		if idHex == "" {
+			idHex = hex.EncodeToString(key)
+		} else if idHex != hex.EncodeToString(key) {
+			log.Println("WARNING: it appears that", opts.DataDir, "was previously initialized with a generated key.")
+			log.Println("The provided id does not match the generated key.")
+			log.Println("\tCurrent value:", hex.EncodeToString(key))
+			log.Println("\tValue passed to postcli:", idHex)
+			if !yes {
+				log.Println("CONTINUING MIGHT ERASE EXISTING DATA. MAKE ABSOLUTELY SURE YOU SPECIFY THE CORRECT VALUE.")
+			}
+			askForConfirmation()
+		}
```

Comment:
I think it would be best to just fail in this situation rather than to allow users to continue. It has no chance of working.